Emby.Server.Implementations/Playlists/PlaylistManager.cs

```diff
@@ -189,41 +189,45 @@ public void AddToPlaylist(string playlistId, IEnumerable<Guid> itemIds, Guid use
 
         private void AddToPlaylistInternal(string playlistId, IEnumerable<Guid> itemIds, User user, DtoOptions options)
         {
-            var playlist = _libraryManager.GetItemById(playlistId) as Playlist;
-
-            if (playlist == null)
-            {
-                throw new ArgumentException("No Playlist exists with the supplied Id");
-            }
-
-            var list = new List<LinkedChild>();
+            // Retrieve the existing playlist
+            var playlist = _libraryManager.GetItemById(playlistId) as Playlist
+                ?? throw new ArgumentException("No Playlist exists with Id " + playlistId);
 
+            // Retrieve all the items to be added to the playlist
             var items = GetPlaylistItems(itemIds, playlist.MediaType, user, options)
                 .Where(i => i.SupportsAddingToPlaylist)
                 .ToList();
 
-            foreach (var item in items)
+            // Remove duplicates from the new items to be added
+            var existingIds = playlist.LinkedChildren.Select(c => c.ItemId).ToHashSet();
+            var uniqueItems = items
+                .Where(i => !existingIds.Contains(i.Id))
+                .GroupBy(i => i.Id)
+                .Select(group => group.First())
+                .ToList();
```

Comment:
The `GroupBy()` and `Select()` here function to remove duplicate entries from the enumerable

Reply:
I would rather we add a newly added item to `existingIds`, rather than going this LINQ/SQL way... would probably be more efficient and easier to understand.

Reply:
What I really wanted to do here was write `.DistinctBy(i => i.Id)` which would be both clear and concise. Do we have somewhere to put Linq extension methods? It's also available in [MoreLINQ](https://github.com/morelinq/MoreLINQ#distinctby), but I don't know if we want to pull in an additional library

Reply:
It's not something I would die on, let it be the way it is for now.
I highly doubt it would _ever_ be a bottleneck 😆